test/validate_resnet50/validation.py

```diff
@@ -0,0 +1,263 @@
+#    3. Add Arguments to take in the backend, the reference log files, the number of iterations/batches,
+#       the data type (real or synthetic)
+#    4. Automate the cloning of benchmarks repo and running the script
+
+validate_with_real_data_command_NG = 'NGRAPH_TF_BACKEND=GPU python tf_cnn_benchmarks.py ' \
+    + '--num_inter_threads=2 --data_format=NCHW --model=resnet50 --batch_size=32 ' \
+    + '--num_gpus=1 --data_dir /mnt/data/TF_ImageNet_latest/ --data_name=imagenet ' \
+    + '--datasets_use_prefetch=False --print_training_accuracy=True ' \
+    + '--num_learning_rate_warmup_epochs=0 --num_batches=100'
+validate_with_real_data_command_TF = 'NGRAPH_TF_DISABLE=1 python tf_cnn_benchmarks.py ' \
+    + '--num_inter_threads=2 --data_format=NHWC --model=resnet50 --batch_size=32 ' \
+    + '--num_gpus=1 --data_dir /mnt/data/TF_ImageNet_latest/ --data_name=imagenet ' \
+    + '--datasets_use_prefetch=False --print_training_accuracy=True ' \
+    + '--num_learning_rate_warmup_epochs=0 --num_batches=100'
+validate_with_synthetic_data_command_NG = 'NGRAPH_TF_BACKEND=GPU python tf_cnn_benchmarks.py ' \
+    + '--num_inter_threads=2 --tf_random_seed=1234 --data_format=NCHW ' \
+    + '--model=resnet50 --batch_size=32 --num_gpus=1 --data_name=imagenet ' \
+    + '--datasets_use_prefetch=False --print_training_accuracy=True ' \
+    + '--num_learning_rate_warmup_epochs=0 --num_batches=100'
+validate_with_synthetic_data_command_TF = 'NGRAPH_TF_DISABLE=1 python tf_cnn_benchmarks.py ' \
+    + '--num_inter_threads=2 --tf_random_seed=1234 --data_format=NHWC --model=resnet50 ' \
+    + '--batch_size=32 --num_gpus=1 --data_name=imagenet --datasets_use_prefetch=False ' \
+    + '--print_training_accuracy=True --num_learning_rate_warmup_epochs=0 --num_batches=100'
+
+
+def command_executor(cmd, verbose=False, msg=None, stdout=None):
+    if verbose or msg is not None:
+        tag = 'Running COMMAND: ' if msg is None else msg
+        print(tag + cmd)
+
+    p = Popen(
+        cmd,
+        shell=True,
+        stdin=PIPE,
+        stdout=PIPE,
+        stderr=PIPE,
+        close_fds=True,
+        bufsize=1)
+    output = p.stdout.read()
+    error_output = p.stderr.read()
+
+    return output, error_output
+
+
+def write_to_file(filename, content):
+    with open(filename, "w") as text_file:
+        text_file.write(content)
+
+
+def parse_training_output(output):
+    to_parse = False
+    total_loss = []
+    top1_acc = []
+    top5_acc = []
+
+    for line in output.strip().split("\n"):
+        if line.split()[0] == 'Step':
+            to_parse = True
+            continue
+
+        elif line.startswith('-----'):
+            to_parse = False
+            continue
+
+        if to_parse:
+            total_loss.append(line.split()[-3])
+            top1_acc.append(line.split()[-2])
+            top5_acc.append(line.split()[-1])
+
+    return total_loss, top1_acc, top5_acc
+
+
+def parse_reference_file(filename):
```

Comment:
`parse_reference_file` and `parse_training_output` can be a single function... I think they are separate because one parses a file, and the other parses string. Maybe we keep the string parsing function and just read the file into a string and reuse.